pkg/dataplane/item.go

```diff
@@ -85,20 +85,26 @@ func (i Item) GetFieldUint64(name string) (uint64, error) {
 }
 
 // For internal use only - DO NOT USE!
-func (i Item) GetShard() (map[int]*ItemChunk, *ItemCurrentChunkMetadata, error) {
+func (i Item) GetShard() (map[uint64]*ItemChunk, *ItemCurrentChunkMetadata, error) {
 	const streamDataPrefix = "__data_stream["
 	const streamMetadataPrefix = "__data_stream_metadata["
 	const offsetPrefix = "__data_stream[0000]["
 
 	currentChunkMetadata := ItemCurrentChunkMetadata{}
-	chunkMap := make(map[int]*ItemChunk)
+	chunkMap := make(map[uint64]*ItemChunk)
 
 	for k, v := range i {
 		if strings.HasPrefix(k, streamDataPrefix) {
-			chunkID, _ := strconv.Atoi(k[len(streamDataPrefix):][:4])
-			offset, _ := strconv.ParseUint(k[len(offsetPrefix):][:16], 16, 64)
-			data, ok := v.([]byte)
-			if !ok {
+			chunkID, ok := strconv.ParseUint(k[len(streamDataPrefix):][:4], 16, 64)
+			if ok != nil {
+				return nil, nil, v3ioerrors.ErrInvalidTypeConversion
+			}
+			offset, ok := strconv.ParseUint(k[len(offsetPrefix):][:16], 16, 64)
+			if ok != nil {
+				return nil, nil, v3ioerrors.ErrInvalidTypeConversion
+			}
+			data, ok2 := v.([]byte)
```

Comment:
`ok2`? Why not just `ok` like in all other places?

Reply:
strong typing..., error type vs bool type @ casting.
is there a way to solve this?

Reply:
just call it `_ok` or `castingSuccess`
`ok2` seems very weird and against convention